tests/unit/core/server/test_base.js

```diff
@@ -451,6 +456,113 @@ describe(__filename, () => {
     });
   });
 
+  describe('TAARDIS', () => {
+    it('dispatches setHashedClientId() if clientId header is present and enableFeatureDiscoTaar is true and app is disco', async () => {
+      const clientId = '1112';
+      const fakeConfig = getFakeConfig({
+        enableFeatureDiscoTaar: true,
+      });
+
+      const { store, sagaMiddleware } = createStoreAndSagas({
+        reducers: {
+          telemetry: telemetryReducer,
+        },
+      });
+
+      const dispatchSpy = sinon.spy(store, 'dispatch');
+
+      const response = await testClient({
+        appInstanceName: 'disco',
+        config: fakeConfig,
+        store,
+        sagaMiddleware,
+      })
+        .get('/en-US/firefox/')
+        .set(DISCO_TAAR_CLIENT_ID_HEADER, clientId)
+        .end();
+
+      sinon.assert.calledWith(dispatchSpy, setHashedClientId(clientId));
+
+      expect(response.statusCode).toEqual(200);
+      expect(store.getState().telemetry.hashedClientId).toEqual(clientId);
+    });
+
+    it('does not dispatch setHashedClientId() if enableFeatureDiscoTaar is false', async () => {
+      const clientId = '1112';
+      const fakeConfig = getFakeConfig({
+        enableFeatureDiscoTaar: false,
+      });
+
+      const { store, sagaMiddleware } = createStoreAndSagas();
+
+      const dispatchSpy = sinon.spy(store, 'dispatch');
+
+      const response = await testClient({
+        appInstanceName: 'disco',
+        config: fakeConfig,
+        store,
+        sagaMiddleware,
+      })
+        .get('/en-US/firefox/')
+        .set(DISCO_TAAR_CLIENT_ID_HEADER, clientId)
+        .end();
+
+      sinon.assert.neverCalledWith(dispatchSpy, setHashedClientId(clientId));
+
+      expect(response.statusCode).toEqual(200);
+      expect(store.getState().telemetry).toEqual(undefined);
+    });
+
+    it('does not dispatch setHashedClientId() if there is no moz-client-id header', async () => {
+      const fakeConfig = getFakeConfig({
+        enableFeatureDiscoTaar: true,
+      });
+      const { store, sagaMiddleware } = createStoreAndSagas();
+
+      const dispatchSpy = sinon.spy(store, 'dispatch');
+
+      const response = await testClient({
+        appInstanceName: 'disco',
+        config: fakeConfig,
+        store,
+        sagaMiddleware,
+      })
+        .get('/en-US/firefox/')
+        .end();
+
+      sinon.assert.neverCalledWith(dispatchSpy, setHashedClientId('1112'));
+
+      expect(response.statusCode).toEqual(200);
+      expect(store.getState().telemetry).toEqual(undefined);
+    });
```

Comment:
We could maybe duplicate this test case for `moz-client-id` set to `null` and `undefined` so that those edge cases are covered too.

Reply:
I've added a case for null but I couldn't set moz-client-id to undefined

Reply:
ok that should be good enough